substrate/primitives/state-machine/src/in_memory_backend.rs

```diff
@@ -21,11 +21,17 @@ use crate::{
 	backend::Backend, trie_backend::TrieBackend, StorageCollection, StorageKey, StorageValue,
 	TrieBackendBuilder,
 };
+use alloc::{collections::BTreeMap, vec::Vec};
 use codec::Codec;
 use hash_db::Hasher;
 use sp_core::storage::{ChildInfo, StateVersion, Storage};
 use sp_trie::{empty_trie_root, LayoutV1, PrefixedMemoryDB};
-use std::collections::{BTreeMap, HashMap};
+
+#[cfg(feature = "std")]
```

Comment:
is it intentional to not use `substrate_runtime`?

Reply:
Yes, because `HashMap` is only available under `std`.

In general this is still quite messy because I only converted a subset of `feature = "std"` into `substrate_runtime`. I had to do some extra changes related to this (and this includes changes to the file you've commented on here) in this PR to make the `full_crypto` feature work again. I will be continuing with more of this refactoring in future PRs.

(Also, a side note, I think we might be able to actually remove the `full_crypto` feature altogether. The whole point of it was to make the code use real crypto libraries as-is instead of calling the host functions when compiled in a `no_std` environment. But once we switch to `substrate_runtime` to detect whether we're compiling a runtime then compiling in `no_std` will by default act like if the `full_crypto` is enabled.)